From 1d8e204562110d47af5762560ae0cb1f76877e8b Mon Sep 17 00:00:00 2001
From: Karthik M <quic_karm@quicinc.com>
Date: Wed, 24 Jan 2024 17:47:04 +0530
Subject: [PATCH] hostapd: WAR for rtnl socket allocation failure

Currently, FT roaming on secondary links in MLD is
not happening when station tries roaming.

When a MLD is created, the hostapd will add the link
addresses to the bridge through rtnl socket. After package
upgrade, rtnl socket allocation is not happening as the bridge
is busy.

Adding WAR by checking the bridge index for rtnl socket creation as the netifd
will add all the interfaces created with "lan" configuration to br-lan

Change-Id: I2cef3d1afaa34e2b21492c61b9e470df50190434
Signed-off-by: Karthik M <quic_karm@quicinc.com>
---
--- a/src/drivers/driver_nl80211.c
+++ b/src/drivers/driver_nl80211.c
@@ -8765,7 +8765,7 @@ static void *i802_init(struct hostapd_da
 	    (params->num_bridge == 0 || !params->bridge[0]))
 		add_ifidx(drv, br_ifindex, drv->ifindex);
 
-	if (bss->added_if_into_bridge || bss->already_in_bridge) {
+	if (params->num_bridge && params->bridge[0]) {
 		int err;
 
 		drv->rtnl_sk = nl_socket_alloc();
-- 
2.17.1

